dd-sdk-android/src/main/cpp/utils/signal-monitor.c

```diff
@@ -0,0 +1,163 @@
+#include "signal-monitor.h"
+#include <errno.h>
+#include <fcntl.h>
+#include <pthread.h>
+#include <signal.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <stdbool.h>
+#include <jni.h>
+#include "../datadog-native-lib.h"
+
+#define HANDLED_SIGNAL_COUNT 6
+
+void handleSignal(int signum, siginfo_t *info,
+                  void *user_context);
+
+bool configureSignalStack(void);
+
+stack_t signalStack;
+/**
+ * Global shared context for Bugsnag reports
+ */
+static JNIEnv *globalEnv;
+/* the signal handler array */
+struct sigaction *globalSigaction;
+
+/* the previous signal handler array */
+struct sigaction *globalSigactionPrevious;
+
+/**
+ * Native signals which will be captured by the Bugsnag signal handler™
+ */
+static const int native_signals[HANDLED_SIGNAL_COUNT + 1] = {
```

Comment:
well here...is the code from bugsnag ...I am not really sure why they are doing this...maybe it is a C++ trick to avoid an OutOfRange exception or something. I wanted also to ask @simonpatarin about this.